README.md

```diff
@@ -1213,6 +1213,94 @@ Here is a complete example for copying bundles from `prod` to `integration`.
    The `--map-version` option adds a specific duration to the version of each
    copied file and bundle. Run `python scripts/copy_bundles --help` for details.
 
+## 7.4 Debugging running lambdas via Pycharm
+
+It's possible to connect a remote debugger to a running lambda function.
+
+1. In PyCharm select `Run > Edit Configuration`, then The **+** to add a new
+   **Python Debug Server**.
+
+   Enter your IP address under **IDE host name**. If you don't have a static IP,
+   leave the **host name** as `localhost` but install https://ngrok.com/. We
+   will use Ngrok to tunnel a URL to the debug server on your machine.
+ 
+   Enter a **Port**, such as `8000`.
+
+   Add the following **Path mappings**:
+
+   local path | remote path
+   --- | ---
+   `<absolute path>/azul/src/azul` | `/var/task/azul`
+   `<absolute path>/azul/.venv/lib/python3.8/site-packages/chalice` | `/var/task/chalice`
+   `<absolute path>/azul/.venv/lib/python3.8/site-packages` | `/opt/python`
+   `<home directory>/.pyenv/versions/3.8.3/lib/python3.8` | `/var/lang/lib/python3.8`
+
+   Depending on which you're debugging, add **one** of the following:
+
+   local path | remote path
+   --- | ---
+   `<absolute path>/azul/lambdas/service/app.py` | `/var/task/app.py`
+   `<absolute path>/azul/lambdas/indexer/app.py` | `/var/task/app.py`
+
+   Copy the `pydevd-pycharm` version` listed in the configurations.
+
+2. Next make some changes.
+
+   Add `pydevd-pycharm==<version from previous step>` to `requirements.txt`.
+
+   Make sure `AZUL_DEBUG` is set to a truthy value in your deployment's
+   `environment.py`.
+   
+   If using Ngrok run:
+
+   ```
+   ngrok tcp 8000
+   ```
+
+   and in your deployment's `environment.py` `AZUL_DEBUG_HOST` and
+   `AZUL_DEBUG_HOST` to the Forwarding URL and port listed by Ngrok. Otherwise
+   set these variables to the values you used in your configurations.
+
+3. Start the debug server, by clicking the debug icon with the configuration
+   selected.
+
+4. Activate the remote debugger in one of two ways:
+
+   * Decorate the route you wish to debug with `RemoteDebugSession()`. Make sure
+     the `RemoteDebugSession()` decorator is applied first:
+     
+     ```
+     @app.route(...)
+     @RemoteDebugSession()
+     def my_route(...):
+         ...
+     ```
+
+   * Use `RemoteDebugSession()` as a context manager around the code you wish to
+     debug.
+
+5. Deploy:
+   ```
+   make package
+   make deploy
+   ```
+
+6. Set breakpoints and trigger the lambda, and debug!
+
+### Warnings / caveats:
+
+* If the debug server isn't running lambdas will hang until they timeout. Make
```

Comment:
"Timeout" is a noun. "Time out" is the verb form. Replace as appropriate throughout these paragraphs.

Reply:
Thanks for pointing that out, that's definitely clarifying!